src/plugin.js

```diff
@@ -435,6 +436,9 @@ const contribAdsPlugin = function(options) {
         },
         nopreroll() {
           this.state = 'content-playback';
+        },
+        contentupdate() {
+          this.state = 'content-set';
```

Comment:
There is a loadstart/contentupdate for the initial source. Is there any guarantee that this isn't that event? Specifically, if the play / adsready came in before loadstart.

Reply:
Good point. I'm not sure there is any such guarantee. We could add a safeguard, something like:
```js
contentupdate() {
  // only change states if this is not the first loadstart/contentupdate
  if (player.ads._hasThereBeenALoadStartDuringPlayerLife) {
    this.state = 'content-set';
  }
}
```

Reply:
You'll need to test it to make sure `_hasThereBeenALoadStartDuringPlayerLife` is set, it's also set on loadstart so the order of event handlers matters here.

Reply:
In hindsight, this check won't have any value for subsequent videos in a playlist since `_hasThereBeenALoadStartDuringPlayerLife` will be true for the life of the player after the first `loadstart`, as the name suggests :P